File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
        
        bool static comp(vector<int> a, vector<int> b){
         return a[1]<b[1];
    }
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        int ans =0;
        sort(intervals.begin(),intervals.end(),comp);
        int i=0,j=1;
        while(j<intervals.size()){
            if(intervals[i][1]>intervals[j][0]){
                j++;
                ans++;
            }
            else{
                i = j;j++;
            }
        }
        return ans;
    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        intervals.sort(key=lambda x:x[1])
        i=0
        ans = 0
        while i<len(intervals):
            j=i+1
            while j<len(intervals) and intervals[i][1]>intervals[j][0]:
                j+=1
                ans+=1
            i=j
        return ans
            
        